processing/src/main/java/io/druid/query/filter/BoundDimFilter.java

```diff
@@ -108,11 +130,14 @@ public boolean isAlphaNumeric()
     byte upperStrictByte = (this.isUpperStrict() == false) ? 0x0 : (byte) 1;
     byte AlphaNumericByte = (this.isAlphaNumeric() == false) ? 0x0 : (byte) 1;
 
+    byte[] extractionFnBytes = extractionFn.getCacheKey();
```

Comment:
@gianm changed that line to have a null check on extractionFn like the rest of the filters, added some cases with extractionFns to the filter cache key tests (e.g., BoundDimFilterTests)